fdbserver/KeyValueStoreMemory.actor.cpp

```diff
@@ -504,56 +521,97 @@ class KeyValueStoreMemory final : public IKeyValueStore, NonCopyable {
 			    cipherKeys.cipherHeaderKey,
 			    getEncryptAuthTokenMode(EncryptAuthTokenMode::ENCRYPT_HEADER_AUTH_TOKEN_MODE_SINGLE),
 			    BlobCipherMetrics::KV_MEMORY);
-			BlobCipherEncryptHeader cipherHeader;
+			uint16_t encryptHeaderSize;
 			Arena arena;
-			StringRef ciphertext =
-			    cipher.encrypt(plaintext, sizeof(int) + v1.size() + v2.size(), &cipherHeader, arena)->toStringRef();
-			log->push(StringRef((const uint8_t*)&cipherHeader, BlobCipherEncryptHeader::headerSize));
-			log->push(ciphertext);
+			if (CLIENT_KNOBS->ENABLE_CONFIGURABLE_ENCRYPTION) {
+				BlobCipherEncryptHeaderRef headerRef;
+				StringRef cipherText = cipher.encrypt(plaintext, v1.size() + v2.size(), &headerRef, arena);
+				Standalone<StringRef> headerRefStr = BlobCipherEncryptHeaderRef::toStringRef(headerRef);
```

Comment:
nit: not related to this PR, but it seems we can avoid `memcpy` by allocating the memory using the same `arena` as underlying `StringBuffer` used by the `IDiskQueue` queue. 

May be we can add a TODO for future optimization if that sounds doable.

Reply:
I can add a TODO for now, IDiskQueue doesn't really allow us to get access to the arena right now so we would need to think of a way to plumb it here. Don't think it makes sense to do it in this pr

Reply:
Second that. It should be done as a separate PR if at all. Thanks